bin/plugin/commands/translations.js

```diff
@@ -0,0 +1,169 @@
+/**
+ * External dependencies
+ */
+const { flatten } = require( 'lodash' );
+const path = require( 'path' );
+const glob = require( 'fast-glob' );
+const fs = require( 'fs' );
+const readline = require( 'readline' );
+
+/**
+ * Internal dependencies
+ */
+const { log, formats } = require( '../lib/logger' );
+const config = require( '../config' );
+
+/**
+ * @typedef WPTranslationsCommandOptions
+ *
+ * @property {string=} directory Optional directory, default is the root `/modules` directory.
+ * @property {string=} output    Optional output PHP file, default is the root `/module-i18n.php` file.
+ */
+
+/**
+ * @typedef WPTranslationsSettings
+ *
+ * @property {string} textdomain Plugin textdomain.
+ * @property {string} directory  Modules directory.
+ * @property {string} output     Output PHP file.
+ */
+
+const options = [
+	{
+		argname: '-d, --directory <directory>',
+		description: 'Modules directory',
+	},
+	{
+		argname: '-d, --output <output>',
+		description: 'Output file',
+	},
+];
+
+/**
+ * Command that generates a PHP file from module header translation strings.
+ *
+ * @param {WPTranslationsCommandOptions} opt
+ */
+async function handler( opt ) {
+	await createTranslations( {
+		textdomain: config.textdomain,
+		directory: opt.directory || 'modules',
+		output: opt.output || 'module-i18n.php',
+	} );
+}
+
+module.exports = {
+	options,
+	handler,
+};
+
+const TAB = '\t';
+const NEWLINE = '\n';
+const FILE_HEADER =
+	[
+		'<?php',
+		'/* THIS IS A GENERATED FILE. DO NOT EDIT DIRECTLY. */',
+		'$generated_i18n_strings = array(',
+	].join( NEWLINE ) + NEWLINE;
```

Comment:
I think we can go with template strings instead due this is executed by Node.

```suggestion
const FILE_HEADER = `<?php
/* THIS IS A GENERATED FILE. DO NOT EDIT DIRECTLY. */
$generated_i18n_strings = array(
`;
```